### PR TITLE
Fix wasNull() in PrestoResultSet

### PR DESCRIPTION
**Pull request by zhenxiao**
Fix https://github.com/facebook/presto/issues/2173

### CODE REVIEW

File: presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoResultSet.java [hunk 1]
```diff
@@ -1700,6 +1704,11 @@ private int columnIndex(String label)
         return index;
     }
 
+    private void setWasNull(Object value)
+    {
+        wasNull.set(value == null ? true : false);
```
Comment (electrum) [inline presto-jdbc/src/main/java/com/facebook/presto/jdbc/PrestoResultSet.java]: The expression `value == null` is already a boolean, so the `? true : false` is redundant (IntelliJ's "Redundant conditional expression" inspection will flag this). So we can simplify that, then inline the method.

File: presto-jdbc/src/test/java/com/facebook/presto/jdbc/TestDriver.java [hunk 2]
```diff
@@ -710,13 +710,18 @@ public void testExecute()
...
+                assertTrue(statement.execute("SELECT 123 x, 'foo' y, NULL z"));
                 ResultSet rs = statement.getResultSet();
                 assertTrue(rs.next());
                 assertEquals(rs.getLong(1), 123);
                 assertEquals(rs.getLong("x"), 123);
+                assertEquals(rs.wasNull(), false);
                 assertEquals(rs.getString(2), "foo");
                 assertEquals(rs.getString("y"), "foo");
+                assertEquals(rs.wasNull(), false);
+                assertEquals(rs.getLong(3), 0);
```
Comment (electrum) [inline presto-jdbc/src/test/java/com/facebook/presto/jdbc/TestDriver.java]: This is probably incorrect since the type of column `z` actually has a unknown type (selecting an untyped literal `NULL` creates a lot of edge cases). We can fix it by casting the `NULL` to a specific type.

File: presto-jdbc/src/test/java/com/facebook/presto/jdbc/TestDriver.java [hunk 3]
```diff
@@ -710,13 +710,18 @@ public void testExecute()
...
+                assertEquals(rs.getObject("z"), null);
+                assertEquals(rs.wasNull(), true);
                 assertFalse(rs.next());
```
Comment (electrum) [inline presto-jdbc/src/test/java/com/facebook/presto/jdbc/TestDriver.java]: We should add a couple more checks: that calling `wasNull()` works for both the ordinal and named getters, and changing the order so it checks `x`, `z`, `y` (so false, true, false).

File: presto-jdbc/src/test/java/com/facebook/presto/jdbc/TestDriver.java [hunk 4]
```diff
@@ -710,13 +710,18 @@ public void testExecute()
     {
         try (Connection connection = createConnection()) {
             try (Statement statement = connection.createStatement()) {
-                assertTrue(statement.execute("SELECT 123 x, 'foo' y"));
+                assertTrue(statement.execute("SELECT 123 x, 'foo' y, NULL z"));
                 ResultSet rs = statement.getResultSet();
                 assertTrue(rs.next());
                 assertEquals(rs.getLong(1), 123);
                 assertEquals(rs.getLong("x"), 123);
+                assertEquals(rs.wasNull(), false);
```
Comment (electrum) [inline presto-jdbc/src/test/java/com/facebook/presto/jdbc/TestDriver.java]: Minor nit: can use `assertTrue` / `assertFalse` for booleans

File: presto-jdbc/src/test/java/com/facebook/presto/jdbc/TestDriver.java [hunk 5]
```diff
@@ -710,13 +710,18 @@ public void testExecute()
...
                 assertEquals(rs.getString(2), "foo");
                 assertEquals(rs.getString("y"), "foo");
+                assertEquals(rs.wasNull(), false);
+                assertEquals(rs.getLong(3), 0);
+                assertEquals(rs.getObject("z"), null);
```
Comment (electrum) [inline presto-jdbc/src/test/java/com/facebook/presto/jdbc/TestDriver.java]: Can also use `assertNull` here

Comment (electrum) [inline presto-jdbc/src/test/java/com/facebook/presto/jdbc/TestDriver.java]: Can use `assertNull` here

**Comment by electrum** [comment]
I fixed up these minor issues and merged. Thanks!
